libvast/include/vast/system/actors.hpp

```diff
@@ -228,22 +227,21 @@ using catalog_actor = typed_actor_fwd<
   // Atomatically replace a set of partititon synopses with another.
   caf::replies_to<atom::replace, std::vector<uuid>,
                   std::vector<augmented_partition_synopsis>>::with<atom::ok>,
-  // Return the candidate partitions for a query.
-  caf::replies_to<atom::candidates, vast::query_context>::with<catalog_result>,
+  // Return the candidate partitions per type for a query.
+  caf::replies_to<atom::candidates, vast::query_context>::with< //
+    std::unordered_map<type, catalog_result>>,
   // Internal telemetry loop.
   caf::reacts_to<atom::telemetry>,
   // Retrieves all known types.
   caf::replies_to<atom::get, atom::type>::with<type_set>,
-  // Registers the given taxonomies.
-  caf::reacts_to<atom::put, taxonomies>,
   // Registers a given layout.
   caf::reacts_to<atom::put, vast::type>,
   // Retrieves the known taxonomies.
   caf::replies_to<atom::get, atom::taxonomies>::with< //
     taxonomies>,
   // Loads the taxonomies on disk.
   caf::replies_to<atom::load>::with< //
-    atom::ok>,
+    taxonomies>,
```

Comment:
I'm not sure about this. At the minimum it needs to be a 2 argument handler that takes `atom::taxonomies` as well.

Reply:
Why exactly? Is the `atom::taxonomies` input then supposed to be an output parameter?

Reply:
So it looks like you actually want to call the `get, taxonomies` handler, maybe the side-effect is not necessary.

Reply:
Considering this `load` handler will be superseded soon when we get rid of the type-registry state file and just use the available partition synopses instead I don't think it makes sense to invest a lot of time here.

Reply:
The `atom::load` handler has been replaced by a method.